src/test/java/pl/kl/weatherservice/location/LocationServiceTest.java

```diff
@@ -0,0 +1,55 @@
+package pl.kl.weatherservice.location;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class LocationServiceTest {
+
+    @Mock
+    LocationRepository locationRepository;
+    @InjectMocks
+    LocationService locationService;
+    @Captor
+    ArgumentCaptor<Location> locationArgumentCaptor;
+
+    @Test
+    void createLocation_thenCreatesNewLocation() {
+        // given
+        when(locationRepository.save(any())).thenReturn(LocationTestHelper.provideLocation());
+        // when
+        locationService.createLocation("Gdansk", "Pomeranian", "Poland", 54.3, 18.6);
+        //then
+        verify(locationRepository).save(locationArgumentCaptor.capture());
+        assertThat(locationArgumentCaptor.getValue().getCity().equals("Gdansk"));
```

Comment:
create a variable for `locationArgumentCaptor.getValue()`

Reply:
Fixed.